### PR TITLE
chore: remove amazon flavor

### LINKED ISSUE #14436: Investigate if `amazon` flavor can be removed
The amazon flavor exists because amazon app store doesn't allow asking the camera permission: https://github.com/ankidroid/Anki-Android/blob/60a823d9c2f7364958b5eaa5a5faee13159b172d/AnkiDroid/src/amazon/AndroidManifest.xml

Since the permission was removed at #14162, investigation is necessary to determine if the flavor can be removed as well, which the main benefit would be decreased release times

### PR DESCRIPTION
**Pull request by BrayanDSO**
unnecessary after cf5624de1824e4d3655a8bbce91a71a5660cc405, since the flavor only existed because we couldn't have the camera permission and now we don't

## Purpose / Description
I thought that the release process was too slow

## Fixes
Closes #14436

## Approach
Removed/commented anything I found that was related to the amazon flavor

## How Has This Been Tested?

Synced gradle and built the app

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/build.gradle [hunk 1]
```diff
@@ -245,12 +238,13 @@ play {
     track.set('alpha')
 }
 
-amazon {
-    securityProfile = file("${homePath}/src/AnkiDroid-Amazon-Publish-Security-Profile.json")
-    applicationId = "amzn1.devportal.mobileapp.524a424d314931494c55383833305539"
-    pathToApks = [ file("./build/outputs/apk/amazon/release/AnkiDroid-amazon-universal-release.apk") ]
-    replaceEdit = true
-}
+// TODO #14161
```
Comment (david-allison) [inline AnkiDroid/build.gradle:241]: Linking for posterity: #14161

**Review by david-allison — COMMENTED** [review]
Also could remove `Permissions::canUseCamera`

**Review by david-allison — APPROVED** [review]

**Comment by mikehardy** [comment]
I won't actually *know* if we can remove it or not until I try to publish it I think. So this one is definitely on me to test, I've got 👀 on it - thank you!

File: AnkiDroid/build.gradle [hunk 2]
```diff
@@ -245,12 +238,13 @@ play {
...
+// amazon {
```
Comment (lukstbit) [inline AnkiDroid/build.gradle:242]: If this is commented out, might as well comment out the amazon publishing plugin _app.brant.amazonappstorepublisher_ in the plugins block.

**Pull request closed without merging**